### PR TITLE
Make msvc_env_cmd no-op for arm64

### PR DESCRIPTION
**Pull request by finnagin**
### Description

Changes `msvc_env_cmd` to output a warning then return an empty string when bits is "arm64".

xref: conda-forge/conda-forge.github.io#1940

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] ~~Add / update necessary tests?~~
- [ ] ~~Add / update outdated documentation?~~

### CODE REVIEW

**Review by isuruf — DISMISSED** [review]

**Review by jezdez — DISMISSED** [review]

**Comment by jezdez** [comment]
pre-commit.ci autofix

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
Sounds like we are waiting on the CLA here

Could someone please share an update on how that is going with (if possible) a rough timeline?

**Comment by finnagin** [comment]
Just wanted to check in again since there hasn't been a response yet to @jakirkham's comment above. 

@jezdez would you be the correct person to ask about clarification on the current status of the CLA? I'm hoping to find an answer to [this question](https://github.com/conda/infrastructure/pull/756#issuecomment-1567096164) if possible.

**Comment by jakirkham** [comment]
Sorry for bumping this again. Just wanted to check if there was an update

File: news/4867-arm64-msvc-env-cmd-no-op [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* Added a check to print an additional warning and return an empty string when bits is "arm64" in msvc_env_cmd
```
Comment (kenodegard) [inline news/4867-arm64-msvc-env-cmd-no-op:19]:
```suggestion
* Added a check to print an additional warning and return an empty string when bits is "arm64" in msvc_env_cmd. (#4867)
```

**Review by kenodegard — APPROVED** [review]

File: conda_build/windows.py [hunk 2]
```diff
@@ -110,6 +110,9 @@ def msvc_env_cmd(bits, config, override=None):
         "If this recipe does not use a compiler, this message is safe to ignore.  "
         "Otherwise, use {{compiler('<language>')}} jinja2 in requirements/build."
     )
+    if bits == "arm64":
+        log.warn("The legacy MSVC compiler setup does not support arm64 builds. ")
+        return ""
```
Comment (isuruf) [inline conda_build/windows.py:115]:
```suggestion
    if bits not in ["64", "32"]:
        log.warn(f"The legacy MSVC compiler setup does not support {bits} builds. ")
        return ""
```

**Review by kenodegard — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**